testplan/testing/multitest/base.py

```diff
@@ -333,6 +335,10 @@ def run_tests(self):
         ctx = [(self.test_context[idx][0], self.test_context[idx][1][:])
                for idx in range(len(self.test_context))]
 
+        if self.cfg.resource_monitor:
```

Comment:
Can we handle other tests like GTest too?

Or if we want finer granularity e.g suite/case level, would current design be easy to adapt?
Maybe do we need to record parent-child relationship via json?

Reply:
Should we have the monitor logic in the Test() base class of all MultiTest, GTest etc?